packages/expo/src/errors/ExpoErrorManager.ts

```diff
@@ -0,0 +1,57 @@
+// Similar interface to the one used in expo modules.
+type CodedError = Error & { code?: string };
+
+let isErrorHandlingEnabled = true;
+let wasHit = false; // whether the original error handler was called
+
+const developmentBuildMessage = `If you're trying to use a module that is not supported in Expo Go, you can create a custom development build. See https://docs.expo.dev/development/introduction/ for more info.`;
+
+function customizeUnavailableMessage(error: CodedError) {
+  error.message += '\n\n' + developmentBuildMessage;
+}
+
+function customizeModuleIsMissingMessage(error: Error) {
+  error.message = `Your JavaScript code tried to access a native module that doesn't exist. 
+
+${developmentBuildMessage}`;
+}
+
+function customizeError(error: Error | CodedError) {
+  if ('code' in error && error.code === 'ERR_UNAVAILABLE') {
+    customizeUnavailableMessage(error);
+  } else if (
+    error.message.includes('Native module cannot be null') || // RN 0.64 and below message
+    error.message.includes('`new NativeEventEmitter()` requires a non-null argument.') // RN 0.65+ message
+  ) {
+    customizeModuleIsMissingMessage(error);
+  }
+}
+
+function errorHandler(originalHandler, error, isFatal) {
+  if (error instanceof Error) {
+    // Suppresses `"main" has not been registered` error only if it was caused by a different error.
+    // Otherwise, we want to show it, in case the user forgot to call `AppRegistry.registerComponent`.
+    if (wasHit && error.message?.includes('has not been registered. This can happen if')) {
+      return;
+    }
```

Comment:
I'm a little worried about outright hiding this error. Is there a way to log it or emit it as a warning so that it's still exposed without clobbering the original error?

Reply:
I did some more testing and found that this piece is unnecessary in Expo Go, as we already prioritize showing other errors besides this one, so I've removed it entirely.